### ISSUE TITLE
what‘s the meaning of the parameter 'sklr'?

### ISSUE THREAD

**Issue by gggllhha** [xingzhehe/AutoLink-Self-supervised-Learning-of-Human-Skeletons-and-Object-Outlines-by-Linking-Keypoints#4]
need to be changed with different datasets?

**Comment by xingzhehe** [comment]
sklr gives larger learning rate for edge weights. With small sklr, the edge
weights converge too slow. I would suggest not to modify this hyper
parameter, i.e., use the same value for all datasets.

ww ***@***.***>于2024年6月5日 周三上午12:11写道：

**Comment by gggllhha** [comment]
Thanks for your reply! And I have another problem when reconstruct the image(256*256) through damaged image and heatmap,the output image is wrong like this.what caused it?
![图片4](https://github.com/xingzhehe/AutoLink-Self-supervised-Learning-of-Human-Skeletons-and-Object-Outlines-by-Linking-Keypoints/assets/45898800/528e3408-95a7-4e2b-a255-bea0d48c3bc6)

**Comment by gggllhha** [comment]
out image after sigmoid still like this 
![图片5](https://github.com/xingzhehe/AutoLink-Self-supervised-Learning-of-Human-Skeletons-and-Object-Outlines-by-Linking-Keypoints/assets/45898800/62b85da5-bf70-4c90-8e0c-1d6667efa85d)

**Comment by xingzhehe** [comment]
> Thanks for your reply! And I have another problem when reconstruct the image(256*256) through damaged image and heatmap,the output image is wrong like this.what caused it? ![图片4](https://private-user-images.githubusercontent.com/45898800/337131982-528e3408-95a7-4e2b-a255-bea0d48c3bc6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2OTIyODUsIm5iZiI6MTcxNzY5MTk4NSwicGF0aCI6Ii80NTg5ODgwMC8zMzcxMzE5ODItNTI4ZTM0MDgtOTVhNy00ZTJiLWEyNTUtYmVhMGQ0OGMzYmM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDE2Mzk0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkyNDBkYjdmNzNmNjMxMWQ0MjVlZmU1YTZlMDRjZjNlMzBlYjYzYzUxZWViYmY2Zjk1YWM2NmMxZDExOTkwNmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rGDDcahWx-ukY1pJxhfLXg7Dzmiqh2Amx9ppPsodDjA)

Did you train the model or you are trying to use the pre-trained model? The pre-trained model only is only trained on 128x128 images, and thus may not be directly applicable to 256x256 images.

**Comment by gggllhha** [comment]
I just use the damaged image and pre-generated keypoints as the input of the decoder, and then calculate the edge map and reconstruct the image. I use this method in another model, not train origin model or use pre-trained model.So is something wrong in skeleton heatmap generation which may be treated as noise when reconstructing image or others?

**Comment by xingzhehe** [comment]
"I just use the damaged image and pre-generated keypoints as the input of the decoder, and then calculate the edge map and reconstruct the image."
Does this mean
1) generate keypoints using the `encoder`
2) calculate the edge map
3) combine edge map with damaged image
4) feed into `decoder`

I have three questions
a) Are the `encoder` and `decoder` pre-trained?
b) at which step, 256x256 is used rather than 128x128?
c) Which steps are included in "another model"?

**Comment by gggllhha** [comment]
Yes, except for key points generation, the rest of the pipeline is the same as in this paper. The way I generate key points is similar to this paper(use hourglass to get heatmap, then generate key points), and the dimensions are the same(batch_size,num_kp,2).The next process is the same as you mentioned above. For the questions, a) encoder and decoder are not pre-trained. b) my input image such as taichi, fashion is 256*256 to get key points, and change the heatmap_size to 256(actually, I try to resize image to 128*128, and other parameter settings are the same as original code, but result still bad). c) except key points generation, other steps are the same as your model. This is my heatmap_overlaid result in the first epoch.
![屏幕截图 2024-06-08 151126](https://github.com/xingzhehe/AutoLink-Self-supervised-Learning-of-Human-Skeletons-and-Object-Outlines-by-Linking-Keypoints/assets/45898800/138a03f1-af99-415a-bee5-495d5d870058)

**Comment by xingzhehe** [comment]
OK in that case... TBH I don't see anything preventing the model learning meaningful things... Are you still using perceptual loss? Does the loss go down at all? The provided output looks like learning in a wrong direction. The untrained model should output something like this

> out image after sigmoid still like this ![图片5](https://private-user-images.githubusercontent.com/45898800/337207115-62b85da5-bf70-4c90-8e0c-1d6667efa85d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc4MzMxMzIsIm5iZiI6MTcxNzgzMjgzMiwicGF0aCI6Ii80NTg5ODgwMC8zMzcyMDcxMTUtNjJiODVkYTUtYmY3MC00YzkwLThlMGMtMWQ2NjY3ZWZhODVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA4VDA3NDcxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1YjU2OGExZmJiY2IxNTk5YThkZWRlNDA1YTlkZDEyNmVkNDQ2NTVhMjJkNmE3NjQ0ZjNiZmI3YThlN2QyZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5Rg0_aAvyBhUdnHvReOuxtxxe8ZPmbJEIWzkFGsR9jw)

**Comment by gggllhha** [comment]
I use perceptual loss, but the problem now is that I can't reconstruct the image like I showed above, so the perceptual loss keeps floating at a certain value and doesn't decrease.

**Comment by xingzhehe** [comment]
Does it decrease at all in the beginning? Or it never decreases? In the
first case, maybe enlarge the decoder? In the second case, there might be
some bug in the code. Do you mind sharing a piece of code?

ww ***@***.***>于2024年6月8日 周六上午1:51写道：

**Comment by gggllhha** [comment]
it never decreases in 20 epoch, because the reconstruct image are always similar. this is some part of encoder、decoder file and loss calculation.
![1](https://github.com/xingzhehe/AutoLink-Self-supervised-Learning-of-Human-Skeletons-and-Object-Outlines-by-Linking-Keypoints/assets/45898800/cc6a8dd0-27e2-4dae-9116-e0657e25ace6)
![2](https://github.com/xingzhehe/AutoLink-Self-supervised-Learning-of-Human-Skeletons-and-Object-Outlines-by-Linking-Keypoints/assets/45898800/5fcaa085-0f43-40db-a320-032366339fdd)
![3](https://github.com/xingzhehe/AutoLink-Self-supervised-Learning-of-Human-Skeletons-and-Object-Outlines-by-Linking-Keypoints/assets/45898800/59cd5d2f-5446-4f58-bf8b-ee17fe1a5130)
![4](https://github.com/xingzhehe/AutoLink-Self-supervised-Learning-of-Human-Skeletons-and-Object-Outlines-by-Linking-Keypoints/assets/45898800/45d4c009-2804-4c33-9a99-b7c12fefab81)

**Comment by xingzhehe** [comment]
OK.. It is so weird. The loss looks like it is not learning anything. Could you change to MSE loss to see if it helps?

**Comment by gggllhha** [comment]
The problem has been solved. There was a bug in the training code. Thank you for your patient reply！
